### ISSUE TITLE
[ISSUE] More test coverage needed for handler type

### ISSUE THREAD

**Issue by benjamin-rood** [mocktools/go-smtp-mock#129]
#### Where?
Defined in https://github.com/mocktools/go-smtp-mock/blob/master/handler.go
and all uses of the `handler` type in `handler_*.go` files

### New issue checklist

- [x] I have updated `smtpmock` to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/mocktools/go-smtp-mock/blob/master/CONTRIBUTING.md)
- [x] I have read the [documentation](https://github.com/mocktools/go-smtp-mock/blob/master/README.md)
- [x] I have searched for [existing GitHub issues](https://github.com/mocktools/go-smtp-mock/issues)

### Issue description

There is insufficient test coverage on `handler` and where it is embedded (e.g. in `handlerRset`). I was able to break it with just one change in https://github.com/mocktools/go-smtp-mock/pull/125 - which was fixed in https://github.com/mocktools/go-smtp-mock/pull/127.
If I could break something like this so easily without a test failing then it's clear we don't have the essential unit test coverage.

**Comment by benjamin-rood** [comment]
@bestwebua Can you assign this to me? I will have time on this weekend to do it.

**Comment by bestwebua** [comment]
Good catch, @benjamin-rood 👍 Seems like good point to add one more file, `handler_test.go` with one test for `handler#clearError()` method: https://github.com/mocktools/go-smtp-mock/blob/34c709d397802ee214a9946e4104c1be84e5adb9/handler.go#L13-L15

Just one straightforward test with `sessionMock` about `session` that exactly recieved the call of `clearError()`. Also we have indirect calls of this one therefore we have 100% for this case.

<img width="827" alt="Screenshot 2022-11-17 at 18 01 07" src="https://user-images.githubusercontent.com/14002692/202509960-9b13fd03-8ef7-4101-83c3-e2b934583e7e.png">

**Comment by bestwebua** [comment]
Seems we have test for this case:

https://github.com/mocktools/go-smtp-mock/blob/1b2ab0e18fb8d75ceb08caf9663cbe175f6a9e57/handler_test.go#L10-L18

We don't need tests for this method, it's my bad, sorry.
